src/Oscoin/API/Client/HTTP.hs

```diff
@@ -51,45 +57,27 @@ data Response = Response
     , responseStatus :: Status
     }
 
-newtype HttpClientT m a = HttpClientT (ReaderT (Requester m) m a)
-    deriving (Functor, Applicative, Monad, MonadMask, MonadCatch, MonadThrow, MonadIO)
-
-instance MonadTrans HttpClientT where
-    lift ma = HttpClientT $ lift ma
-
-instance ( Buildable (Hash c)
-         , Serialise (BlockHash c)
-         , Serialise (PublicKey c)
-         , Serialise (Signature c)
-         , Monad m
-         , MonadIO m
-         ) => MonadClient c (HttpClientT m) where
-    submitTransaction tx =
-        post "/transactions" tx
-
-    getTransaction txId =
-        get $ "/transactions/" <> toUrlPiece (fromHashed txId)
+-- | Make a 'Client' from a function executing the underlying HTTP
+-- requests.
+makeHttpClient
+    :: ( Buildable (Hash c)
+       , Serialise (BlockHash c)
+       , Serialise (PublicKey c)
+       , Serialise (Signature c)
+       , Monad m
+       )
+    => (Request -> m Response) -> Client c m
+makeHttpClient httpRequest = Client{..}
```

Comment:
Perhaps we can call this `httpClientFromRequest`, to differentiate it from the other function, as they sound like they do the same thing currently.

Reply:
Done